pkg/controller/jobframework/reconciler.go

```diff
@@ -473,12 +471,8 @@ func (r *JobReconciler) constructWorkload(ctx context.Context, job GenericJob, o
 		},
 	}
 
-	if utf8.RuneCountInString(job.Object().GetName()) < 64 {
-		wl.Labels[controllerconsts.ParentNameLabel] = job.Object().GetName()
-	}
-
-	if utf8.RuneCountInString(jobGVK.GroupKind().String()) < 64 {
-		wl.Labels[controllerconsts.ParentGroupKindLabel] = jobGVK.GroupKind().String()
+	if uid := string(job.Object().GetUID()); utf8.RuneCountInString(uid) < 64 && uid != "" {
```

Comment:
k8s validation just uses `len`
https://github.com/kubernetes/kubernetes/blob/2c6c4566eff972d6c1320b5f8ad795f88c822d09/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L169

I think using the rune len would be weaker.

I'm wondering if we should just try to put the label and let the apiserver return an error. Note that there is a regex check, so that could potentially fail as well.

Reply:
> I'm wondering if we should just try to put the label and let the apiserver return an error.

We can do that, but I would rather create a workload without a `job-uid` label, since this functionality is not essential for the controller. We could also log a warning if a label can't be added to a workload. Please let me know which of those three options is the best in your opinion.

> Note that there is a regex check, so that could potentially fail as well.

According to [this page](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids), UIDs are standardized as ISO/IEC 9834-8 and as ITU-T X.667. And ISO/IEC 9834-8 says the following:


_UUIDs forming a component of an OID are represented in ASN.1 value notation as the decimal representation of their integer value, but for all other display purposes it is more usual to represent them with hexadecimal digits with a hyphen separating the different fields within the 16-octet UUID._


Regex that kuberentes is using for label validation is the following `(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?`, if I'm not mistaken. This regex should match the UID format, so I think we're good here.

Reply:
> k8s validation just uses len

Replaced RuneCountInString call with len call in 866a09

Reply:
> We can do that, but I would rather create a workload without a job-uid label, since this functionality is not essential for the controller.

In that case, use `IsValidLabelValue` from https://github.com/kubernetes/kubernetes/blob/2c6c4566eff972d6c1320b5f8ad795f88c822d09/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L167

>  We could also log a warning if a label can't be added to a workload.

Logging is not visible to users. Webhooks support warnings https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#best-practices-and-warnings:~:text=Admission%20webhooks%20can%20optionally%20return%20warning%20messages%20that%20are%20returned%20to%20the%20requesting%20client%20in%20HTTP%20Warning%20headers%20with%20a%20warning%20code%20of%20299.%20Warnings%20can%20be%20sent%20with%20allowed%20or%20rejected%20admission%20responses.

But in our case, it's not end users creating Workloads, but users. Maybe a log is not so bad so administrators can take action.

> UIDs are standardized as ISO/IEC 9834-8 and as ITU-T X.667.

I think there is a way where an end user can override the UID and potentially use something that wouldn't pass label validation. However this should be very uncommon. 
The log should be enough :)

Reply:
Replaced len label check with IsValidLabelValue, and added the log message in 0350ad8d.